src/app/components/activity/feed/feed.ts

```diff
@@ -208,6 +209,38 @@ export default class AppActivityFeed extends Vue {
 		this.emitRejectPost(eventItem, community);
 	}
 
+	onPostPinned(eventItem: EventItem) {
+		// Pin the passed in item, and unpin all others.
+		for (const item of this.feed.items) {
+			if (
+				item.feedItem instanceof EventItem &&
+				item.feedItem.type === EventItem.TYPE_POST_ADD &&
+				item.feedItem.action instanceof FiresidePost
+			) {
+				item.feedItem.action.is_pinned = false;
+			}
+		}
+
+		if (
+			eventItem.type === EventItem.TYPE_POST_ADD &&
+			eventItem.action instanceof FiresidePost
+		) {
+			eventItem.action.is_pinned = true;
+		}
+
+		this.onPostEdited(eventItem);
```

Comment:
Not sure what I think about reemitting a pin as an edit post. Maybe we shouldn't emit anything for now.